active/0000-bicep-config-inline-overrides.md

```diff
@@ -0,0 +1,249 @@
+Start Date: 2025-12-08
+---
+
+# Inline bicepconfig overrides via CLI 
+
+## Summary
+
+Introduce ability to override bicepconfig settings inline for the Bicep CLI commands
+
+## Motivation
+
+Internally, we're working on enhancing external input experience by introducing a plugin-based protocol to help shift errors left and provide users with richer type validation when authoring and when validating their Microsoft-internal deployment artifacts.
+There are two distinct scenarios which we need to support:
+1. Validation during authoring in VSCode
+2. Validation when invoking Microsoft-internal tooling SDK/cmdlets which will internally invoke the Bicep CLI to compile bicep artifacts
+
+For scenario (1), users will provide validation parameters in their bicepconfig.json that the plugin requires to successfully resolve an external input,
+```json5
+{
+    "externalInputResolverConfig": {
+        "foo.*": { // external input kind
+            "target": "path-to-plugin-binary",
+            "parameters": { // parameters to be sent over to the plugin
+                "serviceGroupRoot": "/path/to/SGR",
+                "rolloutSpecs": [ "**/*/RolloutSpec.json" ],
+                "servicePresence": [
+                    {
+                        "rolloutInfra": "test",
+                        "regions": [ "eastus", "westus" ]
+                    }
+                ]
+            }
+        }
+    }
+}
+```
+For scenario (2), users will provide the validation parameters via the Microsoft-internal SDK/cmdlet, like so:
+```ps1
+Test-AzureArtifacts `
+    -ServiceGroupRoot "/path/to/SGR"
+    -RolloutSpec "RolloutSpec.json"
+    -RolloutInfra "prod"
+    -Select "regions(australiaeast)"
+    -ConfigurationOverrides '{"Settings": { "ReleaseId": "release-a.b.c" } }'
+```
+
+The implication here is that in scenario (2), the parameters provided via the cmdlet/SDK are likely to be different from the parameters provided in the bicepconfig.json. If the cmdlet-provided-parameters are not taken into account, this could cause confusion when we emit validation diagnostics.
+
+Additionally, there are parameters such as `ConfigurationOverrides` that only make sense to be provided via the cmdlet/SDK and not on the bicepconfig.json, since they are typically provided in a CI pipeline context. Such parameters should similarly be taken into account when performing external input validation.
+
+This spec aims to introduce an override mechanism for bicepconfig.json, and by so doing, the Microsoft-internal cmdlet/SDK would be able to override the `externalInputResolverConfig` parameters when invoking the Bicep CLI with the parameters provided by the user to the cmdlet/SDK.
+
+## Detailed design
+
+### Syntax
+Introduce an additional parameter to the CLI commands `--config-override` that takes in a JSON object:
+
+**Example:**
+```sh
+bicep build --config-override '{"analyzers":{"core":{"enabled":false}},"experimentalFeatures":["lambda"]}'
```

Comment:
Bicep discussion (12/15): Also allow `--config-override` to take a json file - gives user easier alternative for complex json strings